deploy/entrypoint.py

```diff
@@ -0,0 +1,251 @@
+        del envs['KOLIBRI_PEX_URL']
+
+    # CASE A: Running the pex at KOLIBRI_PEX_URL
+    if 'KOLIBRI_PEX_URL' in envs and 'KOLIBRIBUILD_PEX_PATH' not in envs:
+        if envs['KOLIBRI_PEX_URL'] == 'default':
+            envs['KOLIBRI_PEX_URL'] = DEFAULT_KOLIBRI_PEX_URL
+            pex_name = 'kolibri-latest.pex'
+        else:
+            pex_name = os.path.basename(envs['KOLIBRI_PEX_URL'].split("?")[0])  # in case ?querystr...
+        envs['DEPLOY_TYPE'] = 'pex'
+        envs['KOLIBRI_PEX_PATH'] = os.path.join(envs['KOLIBRI_HOME'], pex_name)
+
+    # CASE B: Running the pex from the /kolibribuild volume
+    elif 'KOLIBRI_PEX_URL' not in envs and 'KOLIBRIBUILD_PEX_PATH' in envs:
+        pex_name = os.path.basename(envs['KOLIBRIBUILD_PEX_PATH'])
+        envs['DEPLOY_TYPE'] = 'pex'
+        envs['KOLIBRI_PEX_PATH'] = os.path.join(envs['KOLIBRI_HOME'], pex_name)
+
+    # CASE C: If no PEX url is spefified, we'll run kolibri from source code
+    else:
+        envs['DEPLOY_TYPE'] = 'source'
+
+    return envs
+
+
+
+# FACILITY CREATION
+################################################################################
+
+def get_kolibri_version(kolibri_cmd):
+    """
+    Calls `kolibri_cmd` (list) to extract version information.
+    The parameter `kolibri_cmd` can be either a kolibri command e.g. ['kolibri']
+    or a Kolibri pex invocation like ['python', 'some.pex'].
+    Returns tuple of ints (major, minor), or (None, None) if verison check fails.
+    """
+    MAJOR_MINOR_PAT = re.compile(r'^(?P<major>\d+)\.(?P<minor>\d+)(\.\d+)?.*')
+    cmd = kolibri_cmd[:] + ['--version']
+    logging.info('Calling cmd {} to get the Kolibri version information.'.format(cmd))
+    cmd_str = ' '.join(cmd)
+    proc = subprocess.Popen(cmd_str, stdout=subprocess.PIPE, shell=True)
+    line = proc.stdout.readline()
+    m = MAJOR_MINOR_PAT.search(line)
+    if m:
+        major, minor = m.groupdict()['major'], m.groupdict()['minor']
+        return int(major), int(minor)
+    else:
+        return None, None
+
+
+def create_facility(kolibri_cmd, envs):
+    """
+    Create the facility so users don't have to go through setup wizard.
+    We must use different appraoch based on Kolibri version:
+      - Kolibri versions in range [0, 0.6)    --> SKIP
+      - Kolibri versions in range [0.6, 0.9)  --> legacy_provisiondevice
+      - Kolibri versions in range [0.9, +     --> provisiondevice
+    """
+    major, minor = get_kolibri_version(kolibri_cmd)
+    if major is None or minor is None:
+        logging.warning('Failed to retrieve Kolibri version. Skipping.')
+        return
+    if major >= 1:
+        provisiondevice(kolibri_cmd, envs)
+    if major == 0:
+        if minor < 6:
+            logging.info('Skipping facility creation step.')
+        if minor >= 6 and minor < 9:
+            legacy_provisiondevice(kolibri_cmd, envs)
+        if minor >= 9:
+            provisiondevice(kolibri_cmd, envs)
+
+def provisiondevice(kolibri_cmd, envs):
```

Comment:
I have a slight concern about using this step in the devserver provisioning, because that will reduce developer eyes on the setup wizard, and the incidental QA that goes with it. Fine with it for all other scenarios.

Reply:
There are three scenarios I can imagine wanting to support

1. spin up a completely fresh server with no content or data (for manual integration testing)
2. update an existing server, and keep the existing data and content (like http://kolibribeta.learningequality.org)
3. spin up one or more servers in a deterministic state, with pre-defined content and data (for automated screenshots, automated integration testing, and automated load testing)